src/splat/segtypes/common/textbin.py

```diff
@@ -83,6 +83,11 @@ def write_asm_contents(self, rom_bytes, f: TextIO):
 
         if sym is not None:
             f.write(f"{asm_label} {sym.name}\n")
+            if self.is_text() and asm_label == ".globl":
+                f.write(f".ent {sym.name}\n")
+                f.write(f"{sym.name}:\n")
+            if not self.is_text() and asm_label == ".globl":
+                f.write(f"{sym.name}:\n")
```

Comment:
What do you think about writing it this way?
```suggestion
            if asm_label == ".globl":
                if self.is_text():
                    f.write(f".ent {sym.name}\n")
                f.write(f"{sym.name}:\n")
```